### PR TITLE
add functions returning a list of affected RSs for revamping unit tests

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**:
This PR adds functions that return a list of replicasets to be enqueued for a pod event. This is part of the effort to revamp replicaset unit tests.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: xref #52118

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by crimsonfaith91** [comment]
Commit 1: remove all `addPod`, `updatePod`, `deletePod`, `enqueueReplicaSet`, and `enqueueReplicaSetAfter` usages (remove dependency on workqueue and syncHandler)

File: pkg/controller/replicaset/replica_set_utils.go [hunk 1]
```diff
@@ -174,3 +177,118 @@ func filterOutCondition(conditions []extensions.ReplicaSetCondition, condType ex
 	}
 	return newConditions
 }
+
+// Obtain all RSs affected by pod creation event. The codes should be similar with addPod(),
+// but return RSs instead of enqueuing them to controller's workqueue.
+func getAffectedRSsByPodCreation(pod *v1.Pod, rsc *ReplicaSetController) []*extensions.ReplicaSet {
```
Comment (enisoc) [inline pkg/controller/replicaset/replica_set_utils.go]: The goal was to extract a useful helper that doesn't need any `ReplicaSetController`. However, I'm now thinking that may not be feasible at this time. The functions we depend on, like `getPodReplicaSets`, are tied too closely with the client code (`ReplicaSetLister`) to let us decouple the client calls from the decision making here. We would have to first refactor the client code itself, which is beyond the scope of our test refactor effort.

Let's focus on the other refactorings for now (like `manageReplicas`) and come back to this if we can think of a better way.

**Comment by crimsonfaith91** [comment]
Agreed with @enisoc that client codes refactoring is required, which is out of scope of the test refactor plan. Closing.

**Pull request closed without merging**